jsonschema/exceptions.py

```diff
@@ -61,14 +61,18 @@ def __repr__(self):
         return "<%s: %r>" % (self.__class__.__name__, self.message)
 
     def __str__(self):
+        return self._formatted_message()
+
+    def _formatted_message(self, formatter=partial(pformat, width=72)):
         essential_for_verbose = (
             self.validator, self.validator_value, self.instance, self.schema,
         )
         if any(m is _unset for m in essential_for_verbose):
             return self.message
 
-        pschema = pprint.pformat(self.schema, width=72)
-        pinstance = pprint.pformat(self.instance, width=72)
+        pschema = formatter(self.schema)
+
+        pinstance = pformat(self.instance, width=72)
```

Comment:
We're only using the formatter for the schema, not the instance here, so there's going to be inconsistency in the output.

Can we add a test that covers the instance too and then fix this to make sure the formatter is used for both?